src/notifier/keep_alive_monitor.py

```diff
@@ -24,21 +27,45 @@ class KeepAliveMonitor:
     receiving keep-alive ping events and can notify the user.
     """
 
-    def __init__(self, config: Optional[dict] = None, thresholds: Optional[dict] = None):
+    def __init__(self, config: ConfigView):
         self._notify_manager = None
+        self.config = config
 
-        self._last_keep_alive = {EventService.HARVESTER: datetime.now()}
-        self._last_keep_alive_threshold_seconds = thresholds or {EventService.HARVESTER: 300}
+        # These will be set by set_services() once known
+        self._last_keep_alive: Dict[EventService, datetime] = {}
+        self._last_keep_alive_threshold_seconds: Dict[EventService, int] = {}
 
         self._ping_url = None
-        if config and config["enable_remote_ping"]:
-            self._ping_url = config["ping_url"]
+        if self.config["enable_remote_ping"].get(bool):
+            self._ping_url = self.config["ping_url"].get()
             logging.info(f"Enabled remote pinging to {self._ping_url}")
 
-        # Infer check period from minimum threshold (arbitrary decision)
+        # Check period will be inferred from minimum threshold of all services.
         # Note that this period defines how often high priority notifications
         # will be re-triggered so < 5 min is not recommended
         self._check_period = float("inf")
+        # No point in starting anything yet since we have no known services.
+        self._is_running = False
```

Comment:
Probably won’t need that once we initialized services from within this method.

Reply:
Yeah, just my brainfart after the refactor. Fixd.